GuessingGame/Service.cs

```diff
@@ -3,71 +3,75 @@
-    {
-    Console.WriteLine("Enter the amount of money to bet:");
-     if(!int.TryParse(Console.ReadLine(),out int amount)){
-        Console.WriteLine("Invalid input. Please enter a valid whole number.");
-        GetAmount();
-     }
-     return amount;
-     }
 
-    public int GetNumber()
-    {
-        Console.Write("Enter the number to guess: ");
-    if(!int.TryParse(Console.ReadLine(),out int number)){
-        Console.WriteLine("Invalid input. Please enter a valid whole number.");
-    }
+        public int MoneyToBet()
+        {
+            Console.WriteLine("Enter the amount of money to bet:");
 
-    return number;
-    }
+            if (!int.TryParse(Console.ReadLine(), out int amount))
+            {
+                Console.WriteLine("Invalid input. Please enter a valid whole number.");
+                MoneyToBet();
+            }
+            return amount;
+        }
+
+        public int GuessNumber()
+        {
+            Console.Write("Enter the number to guess: ");
+            if (!int.TryParse(Console.ReadLine(), out int number))
+            {
+                Console.WriteLine("Invalid input. Please enter a valid whole number.");
+            }
 
-    public string GetLevel(){
-        Console.Write(" Choose level (Easy/Medium/Hard) : ");
-        string difficulty = Console.ReadLine();
-        
-                    switch (difficulty.ToLower())
-                    {
-                    case "easy":
-                    return GetResult(1,5,5);
-                    case "medium":
-                    return GetResult(1,10,10);  
-                    case "hard":
-                    return GetResult(1,15,15);
-                    default:
+            return number;
+        }
+
+        public string ChooseDifficultyLevelWhichReturnMinMaxMutipler(int amount,int guessNumber)
+        {
+            Console.Write(" Choose level (Easy/Medium/Hard) : ");
+            string difficulty = Console.ReadLine();
+
+            switch (difficulty.ToLower())
+            {
+                case "easy":
+                    return GuessingGameResult(1, 5, 5,amount,guessNumber);
+                case "medium":
+                    return GuessingGameResult(1, 10, 10,amount,guessNumber);
+                case "hard":
+                    return GuessingGameResult(1, 15, 15,amount,guessNumber);
+                default:
                     Console.WriteLine("enter a correct string");
-                        return GetLevel();
-                       
-                    }
+                    return ChooseDifficultyLevelWhichReturnMinMaxMutipler(amount,guessNumber);
+            }
 
-    }
-    public string GetResult(int minRange,int maxRange,int multiplier){
-                    Random random = new Random();
-                    int randomNumber = random.Next(minRange, maxRange);
+        }
+        public string GuessingGameResult(int minRange, int maxRange, int multiplier,int amount,int guessNumber)
```

Comment:
Good refactor, adding the guessNumber as a parameter has made you code more testable.